starfish/pipeline/features/spots/detector/_base.py

```diff
@@ -1,9 +1,30 @@
+from typing import Union
+
+import xarray as xr
+import numpy as np
+
 from starfish.image import ImageStack
 from starfish.pipeline.algorithmbase import AlgorithmBase
+from starfish.pipeline.features.spot_attributes import SpotAttributes
+from starfish.intensity_table import IntensityTable
 
 
 class SpotFinderAlgorithmBase(AlgorithmBase):
```

Comment:
Adds typing and two methods that are general to all implementing algorithms: `image_to_spots`, which finds spots and emits a spot_attributes table, and `_get_measurement_function` which converts `max`, `mean` etc to the corresponding numpy function.